spiralgo/ecommserver
# Running Elastic On Localhost.

In order to ditch Google Cloud SQL and Elastic Cloud dependencies at least on the development environment, we can set up Elastic Search and Kibana on localhost.

### Installing Elasticsearch and Kibana:

- The reviewer should install Elasticsearch and its viewer tool Kibana first.
- The reviewer should unzip the files in a disk where there is enough disk space (more than 50 GB would be fine, even if it will never use that much space,  otherwise there may occur some problems related to the default setting of Elastic.)
- The reviewer should apply the required steps on the links below:
https://www.elastic.co/downloads/elasticsearch
https://www.elastic.co/downloads/kibana

### Updating `launch.json`:

```java
	   "env": {
				"DATABASE_USER":"root",
				"DATABASE_PASSWORD":"root",
				"DATABASE_NAME":"ecommDB",
                                "ELASTIC_URIS":"localhost:9200",
                                "ELASTIC_USERNAME": "elastic",
                                "ELASTIC_USING_SSL":"false",
				"GCP_ENABLED":"false",
				"SPRING_ACTIVE_PROFILE":"default,dev"
			}

```

### Running https://github.com/spiralgo/mysql-es-synch project on Docker:

- The reviewer should make sure you have the latest code by pulling from GitHub.
- Should run this command to run the container:
  `docker compose up`


### TO TEST:

 - The reviewer should create a new item and check if Kibana has the item. 
 (On Kibana-> Discover. In the first attempt, Kibana will want creating an `index pattern`
  As our index is `item_idx`, we can create an `index pattern` with the same name:  `item_idx`)
